### ISSUE TITLE
Change SwiftCheck dependency as private

### ISSUE THREAD

**Issue by inamiy** [thoughtbot/Runes#106]
It seems `SwiftCheck` has been introduced since `v4.2.2`, and it is affecting end users to always pull its unnecessary repository via Swift Package Manager:

```bash
$ cd my-project-using-runes/
$ swift build
Updating https://github.com/thoughtbot/Runes.git
Fetching git@github.com:typelift/SwiftCheck.git
...
```

I have tried both Xcode 10 (Swift 5.0) and Xcode 11 Beta (Swift 5.1), but result was the same.
It's probably due to Swift Package Manager still not able to handle test dependencies nicely.

I think there needs some workaround e.g. https://github.com/pointfreeco/swift-snapshot-testing/issues/201 .
What do you think?

**Comment by gfontenot** [comment]
This is definitely a Swift Package Manager deficiency, but is it causing problems for you? It should be downloaded, but it should _not_ be built. I’m a little hesitant to add custom behavior to work around issues with individual dependency managers, tbh.

**Comment by inamiy** [comment]
I personally found redundant download a bit painful, but since `SwiftCheck` is not a very large repository (currently) and has no other public dependencies, I can barely live with it.

But it's really up to the owner of the dependency who may suddenly grow its repo size or create another dependency graph huge.
So, since Runes is popular, IMO it's better to avoid such scenario beforehand.

P.S. 
Last time I had download-time issue in pointfreeco/swift-snapshot-testing#201 which took like several minutes that I couldn't tolerate.
(But plz feel free to close this issue if you think it will not happen in your repo)

**Comment by gfontenot** [comment]
Looked into this a bit more just now. Some thoughts:

 - It looks like [this proposal](https://github.com/apple/swift-evolution/blob/master/proposals/0226-package-manager-target-based-dep-resolution.md) is intended to solve this issue, right?
 - The fix applied by the Point Free folks relies on their use of `make` for generating the project, which isn't something we're currently doing and isn't something I'm particularly interested in setting up/maintaining in the long term.
 - The fix applied by Quick was only used for the 3.1 tooling, and has since been abandoned in the most recent Package file.
 - It looks like that might be because the specified env variable isn't set anymore? I actually don't see anything in the process env that relates to Swift at all, so I'm not sure how to use a similar technique today.

It seems to me like the only realistic fix for this issue today is to avoid test dependencies entirely, which means dropping support for `swift test`. This would really bum me out, but maybe it's worth it for the sake of consuming code?

**Comment by gfontenot** [comment]
I just added support for Swift 5.2 which fixes this issue!
